frontend/packages/knative-plugin/src/components/overview/KnativeServingDetailsPage.tsx

```diff
@@ -4,11 +4,16 @@ import { DetailsForKind } from '@console/internal/components/default-resource';
 import { DetailsPage } from '@console/internal/components/factory';
 import { navFactory } from '@console/internal/components/utils';
 import { K8sResourceKindReference, referenceForModel } from '@console/internal/module/k8s';
-import { KnativeServingModel } from '../../models';
-
-const knativeServingReference: K8sResourceKindReference = referenceForModel(KnativeServingModel);
+import { useK8sModel } from '@console/shared/src/hooks/useK8sModel';
+import { KnativeServingModel, KnativeServingModelV1Beta1 } from '../../models';
 
 const KnativeServingDetailsPage: React.FC<React.ComponentProps<typeof DetailsPage>> = (props) => {
+  const [model] = useK8sModel(props.kind);
+
+  const knativeServingReference: K8sResourceKindReference =
```

Comment:
can make a comment on the purpose of doing a check with `v1alpha1` and another version